### PR TITLE
added book enity 

### PR DESCRIPTION
**Pull request by volodymyrruso**
HW #1

## Summary by CodeRabbit

- **New Features**
	- Introduced a new Book entity with attributes for enhanced book management.
	- Launched a Book repository for efficient data handling and retrieval.
	- Implemented new service methods to save and retrieve books, improving user interaction with the application.

### CODE REVIEW

File: src/main/java/org/mate/bookstorespringboot/service/interfaces/BookService.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+     * @return       the saved book
+     */
+    Book save(Book book);
+
+    /**
+     * Find all books.
+     *
+     * @return         list of all books
+     */
+    List<Book> findAll();
```
Comment (liliia-ponomarenko) [inline src/main/java/org/mate/bookstorespringboot/service/interfaces/BookService.java]: I guess that java-doc here is redundant;)

File: src/main/java/org/mate/bookstorespringboot/model/Book.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NonNull
+    @Column(name = "title")
+    private String title;
+
+    @NonNull
+    @Column(name = "author")
```
Comment (liliia-ponomarenko) [inline src/main/java/org/mate/bookstorespringboot/model/Book.java]: you can use the column parameter `nullable = false` instead annotation;)

**Review by liliia-ponomarenko — APPROVED** [review]
Great! 😊 A few recommendation ;)

File: src/main/java/org/mate/bookstorespringboot/service/interfaces/BookService.java (hunk 1 above)
Comment (volodymyrruso) [reply src/main/java/org/mate/bookstorespringboot/service/interfaces/BookService.java]: oh okay, I just got used to adding the javadoc to any interfaces, no matter how obvious the method is :) done!

**Pull request merged**